sdk/core/azure-core/inc/azure/core/uuid.hpp

```diff
@@ -39,10 +40,23 @@ namespace Azure { namespace Core {
      */
     std::string ToString();
 
+    /**
+     * @brief Returns the binary value of the Uuid for consumption by clients who need non-string
+     * representation of the Uuid
+     * @returns An array with the binary representation of the Uuid.
+     */
+    std::array<uint8_t, UuidSize> const& AsArray() const { return m_uuid; }
+
     /**
      * @brief Creates a new random UUID.
      *
      */
     static Uuid CreateUuid();
+
+    /**
+     * @brief Construct a Uuid from an existing UUID represented as an array of bytes.
+     * @details Creates a Uuid from a UUID created in an external scope.
+     */
+    static Uuid FromArray(std::array<uint8_t, UuidSize> const& uuid);
```

Comment:
I think we have a naming pattern for factories to prefix with a verb like `Create`.

`CreateFromArray` is clear that this is a ctor.